pkg/repository/v1manifest/manifest.go

```diff
@@ -95,31 +95,50 @@ var ManifestsConfig = map[string]ty{
 
 var knownVersions = map[string]struct{}{"0.1.0": {}}
 
-// verifySignature ensures that each signature in manifest::signatures is a valid signature of manifest::signed.
-func (manifest *Manifest) verifySignature(keys crypto.KeyStore) error {
+// VerifySignature ensures that threshold signature in manifest::signatures is a valid signature of manifest::signed.
+func (manifest *Manifest) VerifySignature(threshold uint, keys crypto.KeyStore) error {
 	if keys == nil {
 		return nil
 	}
 
+	if threshold == 0 {
+		return nil
+	}
+
+	if len(manifest.Signatures) < int(threshold) {
+		return errors.Errorf("only %d signature but need %d", len(manifest.Signatures), threshold)
+	}
+
 	payload, err := cjson.Marshal(manifest.Signed)
 	if err != nil {
 		return nil
 	}
 
+	var validCount uint
+	var errs []error
 	for _, sig := range manifest.Signatures {
 		// TODO check that KeyID belongs to a role which is authorised to sign this manifest
 		key := keys.Get(sig.KeyID)
 		if key == nil {
 			// TODO use SignatureError
-			return fmt.Errorf("signature key %s not found", sig.KeyID)
+			err := fmt.Errorf("signature key %s not found", sig.KeyID)
+			errs = append(errs, err)
+			continue
 		}
-		if err := key.VerifySignature(payload, sig.Sig); err != nil {
-			// TODO use SignatureError
-			return err
+		err := key.VerifySignature(payload, sig.Sig)
+		// TODO use SignatureError
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			validCount++
 		}
 	}
 
-	return nil
+	if validCount >= threshold {
```

Comment:
I think all signatures must be correct, not just the threshold number. Threshold just dictates the number of signatures required.